spharpy/classes/audio.py

```diff
@@ -1,12 +1,280 @@
+        ``'maxN'``, ``'SN3D'`` or ``'SNM'``. (maxN is only supported up
+        to 3rd order)
+    channel_convention : str
+        Channel ordering convention, either ``'ACN'`` or ``'FuMa'``.
+        (FuMa is only supported up to 3rd order)
+    condon_shortley : bool
+        Flag to indicate if the Condon-Shortley phase term is included
+        (``True``) or not (``False``).
+    comment : str
+        A comment related to `data`. The default is ``""``.
+    """
+
+    def __init__(self, data, frequencies, basis_type, normalization,
+                 channel_convention, condon_shortley, comment=""):
+
+        self._normalization = normalization
+        self._channel_convention = channel_convention
+
+        FrequencyData.__init__(self, data=data, frequencies=frequencies,
+                               comment=comment)
+        _SphericalHarmonicAudio.__init__(
+            self, basis_type, normalization, channel_convention,
+            condon_shortley)
+
+    @property
+    def freq(self):
+        """Return or set the data in the frequency domain."""
+        data = FrequencyData.freq.fget(self)
+
+        # renormalize according to desired normalization
+        if not self.normalization == "N3D":
+            data = renormalize(data,
+                               "ACN",
+                               "N3D",
+                               self.normalization,
+                               axis=-2)
+        # change channel convention according to desired convention
+        if not self.channel_convention == "ACN":
+            data = change_channel_convention(
+                        data,
+                        "ACN",
+                        self.channel_convention,
+                        axis=-2)
+        return data
+
+    @freq.setter
+    def freq(self, value):
+        """Return or set the data in the frequency domain."""
+        if len(value.shape) < 3:
+            raise ValueError("Invalid number of dimensions. Data should have "
+                             "at least 3 dimensions.")
+
+        # check dimensions
+        if len(value.shape) < 3:
+            raise ValueError("Invalid number of dimensions. Data should have "
+                             "at least 3 dimensions.")
+
+        # convert to N3D and ACN if necessary
+        if not self.normalization == "N3D":
+            value = renormalize(value,
+                                self.channel_convention,
+                                self.normalization,
+                                "N3D",
+                                axis=-2)
+        if not self.channel_convention == "ACN":
+            value = change_channel_convention(
+                        value,
+                        self.channel_convention,
+                        "ACN",
+                        axis=-2)
+
+        FrequencyData.freq.fset(self, value)
+
+
+class SphericalHarmonicSignal(_SphericalHarmonicAudio, Signal):
```

Comment:
Maybe I do not fully get the concept, but shouldn't this be
```suggestion
class SphericalHarmonicSignal(SphericalHarmonicTimeData, SphericalHarmonicFrequencyData):
```